src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,150 @@
+    keys: string
+    actionArguments: number
+  }
+}
+
+@Component({
+  selector: 'sofie-edit-action-triggers',
+  templateUrl: './edit-action-triggers.component.html',
+  styleUrls: ['./edit-action-triggers.component.scss'],
+})
+export class EditActionTriggersComponent implements OnChanges, OnInit {
+  @Output() private readonly cancelActionTrigger: EventEmitter<void> = new EventEmitter<void>()
+  @Input() public selectedActionTrigger: ActionTrigger<KeyboardAndSelectionTriggerData> | null
+  public submitting: boolean = false
+  public submitted: boolean = false
+  public actions: Tv2PartAction[]
+  public loading: boolean = false
+  public keyPress: boolean = false
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly actionTriggerService: ActionTriggerService,
+    private readonly logger: Logger,
+    private readonly actionStateService: ActionStateService
+  ) {}
+
+  public actionForm: UntypedFormGroup = this.fb.group({
+    actionId: ['', [Validators.required]],
+    data: this.fb.group({
+      keys: ['', [Validators.required]],
+      actionArguments: '',
+    }),
+  })
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const actionChange: SimpleChange | undefined = changes['selectedActionTrigger']
+    if (actionChange) {
+      const action: ActionTrigger<KeyboardAndSelectionTriggerData> | null = actionChange.currentValue
+      if (action) {
+        this.patchValue(action)
+      } else {
+        this.actionForm.reset()
+      }
+    }
+  }
+
+  public ngOnInit(): void {
+    this.loading = true
+    this.actionStateService
+      .subscribeToRundownActions('jSXbtcsHTPjebGXurMzP401Z3u0_')
+      .then(observable => observable.subscribe(actions => this.onActionsChanged(actions as Tv2PartAction[])))
+      .catch(error => this.logger.data(error).error('Error while listening to Action events'))
+  }
+
+  private onActionsChanged(loadedActions: Tv2PartAction[]): void {
+    this.actions = loadedActions
+    this.loading = false
+  }
+
+  private patchValue(actionValue: ActionTrigger<KeyboardAndSelectionTriggerData>): void {
+    this.actionForm.patchValue({ actionId: actionValue.actionId, data: { keys: actionValue.data.keys.join(' + '), actionArguments: actionValue.data.actionArguments } })
+  }
+
+  public onKeyDown(event: KeyboardEvent): void {
+    event.preventDefault()
+    const newKeyCode: string = SHORTCUT_KEYS_MAPPINGS[event.code] ? SHORTCUT_KEYS_MAPPINGS[event.code] : event.code
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value.split(' + ') : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
+        currentKeys.push(newKeyCode)
+        this.actionForm?.get('data')?.get('keys')?.patchValue(currentKeys.join(' + '))
+      }
+    } else {
+      this.actionForm?.get('data')?.get('keys')?.patchValue(newKeyCode)
+      this.keyPress = true
+    }
```

Comment:
By utilizing `return` you can get rid of a layer of nesting :)
```
public onKeyDown(...): void {
  ...
  if (!this.keyPress) {
    this.actionForm?.get(...)...
    this.keyPress = true
    return
  }
  const currentKeys: string[] = ...
  if (currentKeys...) {
    ...
  }
}
```

Reply:
Done